ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -817,7 +820,21 @@ impl<'a> X64CodeGen<'a> {
         let mut locs: Vec<LocalAlloc> = Vec::new();
         let gi = inst.guard_info(self.m);
         for lidx in gi.lives() {
-            locs.push(*self.ra.allocation(*lidx));
+            match self.m.inst(*lidx) {
+                jit_ir::Inst::ProxyConst(c) => {
```

Comment:
When we see a guard with a constant, we can just not emit the guard entirely. [The optimiser will remove such guards, but if optimisations are turned off, we will get such guards. But there's no point codegenning them.] That will simplify this method a lot I think!

Reply:
This is not the condition of the guard. It's a live variable that needs to be tracked and deoptimised.

Reply:
Ah!